### PR TITLE
protect against double inserting middlewares

### PR DESCRIPTION
**Pull request by jcarres-mdsol**
It seems that mistake do happen and does not look nice. 
Thoughts on the approach? I could not find a way to find out at initialization time.

@ykitamura-mdsol  @jfeltesse-mdsol

### CODE REVIEW

**Comment by ykitamura-mdsol** [comment]
Why not just fix the mistake?

**Comment by jfeltesse-mdsol** [comment]
is this to prevent end users from shooting themselves in the foot by adding the middleware twice in their chain?

**Comment by jcarres-mdsol** [comment]
Right, is to prevent users from adding it twice.
I've seen this happen in one instance, could fix it there also and there is an argument for doing that instead. But on the other hand do not want to play a whac-a-mole game. What do you think?

File: spec/lib/excon/zipkin-tracer_spec.rb [hunk 1]
```diff
@@ -15,6 +15,7 @@ def process(body, url, headers = {})
                           )
     connection.request
 
+    return headers if !headers.empty?
```
Comment (jfeltesse-mdsol) [inline spec/lib/excon/zipkin-tracer_spec.rb]: `unless headers.empty?`

File: lib/zipkin-tracer/faraday/zipkin-tracer.rb [hunk 2]
```diff
@@ -10,6 +10,8 @@ def initialize(app, service_name = nil)
     end
 
     def call(env)
+      # Prevent calling this middleware twice
+      return @app.call(env) if !env[:request_headers][b3_headers().values.first].nil?
```
Comment (jfeltesse-mdsol) [inline lib/zipkin-tracer/faraday/zipkin-tracer.rb]: - How about printing some warning in the "called twice" case? Silently returning means the user will never know their config is dodgy.
- Code-wise how about making the headers names constants and simply pick one directly, e.g. 
  `return @app.call(env) if env[:request_headers].key?(X-B3-TRACEID)`. 
  The current implementation here will create a new array from the hash values on every single call, even when the config is sane.

**Pull request closed without merging**